doc/glossary.rst

```diff
@@ -285,6 +285,16 @@ General Concepts
         Note that in this case, the precision can be platform dependent.
         The `numeric` dtype refers to accepting both `integer` and `floating`.
 
+        In regards to the efficiency and precision of the NumPy array, the data type
+        (`dtype`) can make a significant impact. If one chooses a data type with
+        a higher precision like `np.float64` or `np.int64` it will result in slow
```

Comment:
I think that we need to mitigate the `slow operations`. Basically, using lower precision can provide a speed up if the operation under the hood leverage some optimizations as vectorization, SIMD, or cache optimization. If those are not implemented or taking care, you might not get any speed up.